src/ParseObject.ts

```diff
@@ -514,6 +514,9 @@ class ParseObject<T extends Attributes = Attributes> {
 
   static _getRequestOptions(options: RequestOptions & FullOptions & { json?: boolean } = {}) {
     const requestOptions: RequestOptions & FullOptions & { json?: boolean } = {};
+    if (!options) {
```

Comment:
We could skip on undefined and throw a nice error message on invalid value:

```suggestion
    if (options === undefined) {
      return requestOptions;
    }
    if (Object.prototype.toString.call(options) !== '[object Object]') {
      throw new Error('request options must be of type Object');
    }
```

Reply:
As I see it the following part has unreachable code so I didn't include it. If `options` us `undefined` it will be set to `{}` by the default function parameter.

```
if (options === undefined) {
      return requestOptions;
}
```